src/utils/endianness.c

```diff
@@ -35,19 +35,29 @@ uint64_t _z_le_load64(const uint8_t *src) {
            ((uint64_t)src[4] << 32) | ((uint64_t)src[5] << 40) | ((uint64_t)src[6] << 48) | ((uint64_t)src[7] << 56);
 }
 
-void _z_le_store16(const uint16_t val, uint8_t *dst) {
+static size_t _z_le_calc_vle_len(size_t size, uint8_t *buff) {
+    size_t vle_len = size;
+    while (vle_len > 1 && buff[vle_len - 1] == 0) {
+        --vle_len;
+    }
+    return vle_len;
+}
+
+size_t _z_le_store16(const uint16_t val, uint8_t *dst) {
     dst[0] = (uint8_t)(val >> 0);
     dst[1] = (uint8_t)(val >> 8);
+    return _z_le_calc_vle_len(2, dst);
 }
 
-void _z_le_store32(const uint32_t val, uint8_t *dst) {
+size_t _z_le_store32(const uint32_t val, uint8_t *dst) {
     dst[0] = (uint8_t)(val >> 0);
```

Comment:
if you are anyway looping over the bytes to count vle length. Maybe it is just simpler just too make bit shifts in a loop and terminate early once val turns to 0 ?
```
for (size_t i = 0; i < sizeof(val); ++i) {
    dst[i] = (uint8_t)val;
    val = val >> 8;
    if (val == 0) return i + 1;
}

This also can be done as a single macro and probably slightly reduce amount of code
```

Reply:
I think a macro would make it less readable in this case. 

Now for combining the loop, I don't know, not that performance really matter but it might be a `strlcpy` situation (https://nrk.neocities.org/articles/cpu-vs-common-sense)

Reply:
I'm not sure strlcpy is related since you'll have to loop once in both cases. And given that pico targets low-end controllers it is more likely that less code packed in a loop will have better performance than an unrolled loop given that vector registers might not be available at all.

Reply:
I suppose to give equal chances we can still copy the whole uint, but check for last non-zero byte in the same loop. It will produce the similar code on smart-enough compiler (if it can optimize checking for last non-zero byte, it should likely be able to optimize a copy of uint), but still keeps the benefit of having smaller code size.